src/main/java/com/letsparty/web/controller/PartyController.java

```diff
@@ -181,34 +191,89 @@ public String modify(@PathVariable int partyNo, @Valid PartyForm partyForm, Bind
 		PostForm postForm = new PostForm();
 		model.addAttribute("postForm", postForm);
-		return "/page/party/post";
+		return "/page/party/post-form";
 	}
 	
 	// 게시물 제출
 	@PreAuthorize("isAuthenticated()")
-	@PostMapping("/{partyNo}/post")
+	@PostMapping("/{partyNo}/post-form")
 	public String addPost(@PathVariable int partyNo, @AuthenticationPrincipal LoginUser loginUser, @Valid PostForm postForm,
 			BindingResult error, Model model) {
 		UserPartyApplication upa = userPartyApplicationService.findByPartyNoAndUserId(partyNo, loginUser.getId());
 		model.addAttribute("upa", upa);
 		// TODO 제목, 본문 글자 수 제한하는 유효성 구현하기
 		if (error.hasErrors()) {
-			return "page/party/post";
+			return "page/party/post-form";
 		}
 		partyService.insertPost(postForm, partyNo, loginUser.getId());
 		return "redirect:/party/{partyNo}";
 	}
 	
 	@GetMapping("/{partyNo}")
-	public String home(@PathVariable int partyNo, Model model) {
-		// partyNo를 사용하여 파티 게시물을 조회합니다.
-		return "page/party/home";
+	public String home(@PathVariable int partyNo) {
+		// partyNo를 사용해서 마지막 게시글 번호를 불러온다.
+		int postNo = postService.getLastPostNoByPartyNo(partyNo);
+		return "redirect:/party/{partyNo}/read/" + postNo;
+	}
+	
+	@GetMapping("/{partyNo}/read/{postNo}")
+	public String read(@PathVariable("partyNo") int partyNo, @PathVariable("postNo") int postNo) {
+		postService.readIncrement(partyNo, postNo);
+		return "redirect:/party/{partyNo}/post/{postNo}";
+	}
+	
+	@GetMapping("/{partyNo}/post/{postNo}")
+	public String readPost(@PathVariable("partyNo") int partyNo, @PathVariable("postNo") int postNo, @AuthenticationPrincipal LoginUser loginUser, Model model) {
+		if (null != loginUser) {
+			User user = userService.getUserByName(loginUser.getId());
+			UserPartyApplication loginUpa = userPartyApplicationService.findByPartyNoAndUserId(partyNo, loginUser.getId());
+			if (null != loginUpa) {
+				model.addAttribute("loginUpa", loginUpa);
+			}
+			model.addAttribute("loginUser", user);
+		}
+		
+		// 해당 게시글 번호로 게시글 정보, 첨부파일 정보들을 불러와 저장한 뒤 화면에 표시한다.
+		Post post = postService.getPostByPostNoAndPartyNo(partyNo, postNo);
+		UserPartyApplication upa = userPartyApplicationService.findByPartyNoAndUserId(partyNo, post.getUser().getId());
+		model.addAttribute("post", post);
+		model.addAttribute("upa", upa);
+		
+		// 게시글의 각종 첨부 파일 내용들(이미지, 동영상, 지도, 투표)
+		PostAttachment pa = mediaService.getMediaByPostId(post.getId());
+		model.addAttribute("pa",pa);
+		
+		// 해당 게시글 번호의 앞뒤로 2개씩 게시글의 제목, 닉네임, 본문, 댓글수, 조회수를 가져와 화면에 표시한다.
+		List<SimplePostDto> spostDto = postService.getSimplePostLimit5(partyNo, postNo);
+		model.addAttribute("simPosts", spostDto);
+		
+		// 마지막 게시글번호가 포함되면 오른쪽 정렬을 하기 위한 로직
+		List<Integer> sPostNos = new ArrayList<>();
+		for (SimplePostDto sPost : spostDto) {
+			sPostNos.add(sPost.getPostNo());
+		}
+		BeginEndPostNo beginEndNo = postService.getBeginAndEndPostNo(partyNo);
+		model.addAttribute("beginEndNo", beginEndNo);
+		model.addAttribute("sPostNos", sPostNos);
```

Comment:
```suggestion
		// 해당 게시글 번호의 앞뒤로 2개씩 게시글의 제목, 닉네임, 본문, 댓글수, 조회수를 가져와 화면에 표시한다.
		List<SimplePostDto> sPostDtos = postService.getSimplePostLimit5(partyNo, postNo);
		model.addAttribute("simPosts", sPostDtos);
		
		// 마지막 게시글번호가 포함되면 오른쪽 정렬을 하기 위한 로직
		BeginEndPostNo beginEndNo = postService.getBeginAndEndPostNo(partyNo);
		boolean hasBeginNo = false;
		boolean hasEndNo = false;
		if (beginEndNo.getBeginNo() != null && beginEndNo.getBeginNo() == spostDto.get(0).getPostNo()) {
    hasBeginNo = true;
}
		if (beginEndNo.getEndNo() != null && beginEndNo.getEndNo() == spostDto.get(spostDto.size() - 1).getPostNo()) {
			hasEndNo = true;
		}
		model.addAttribute("hasBeginNo", hasBeginNo);
		model.addAttribute("hasEndNo", hasEndNo);
```

덩치 큰 List를 만들기보다 boolean 두 개를 빠른 로직으로 계산해 model로 넘기는 것을 제안합니다.
그리고 글이 전혀 없어 getBeginAndEndPostNo 반환값이 null인 경우를 고려해보세요.

Reply:
반환값이 null 안나오게 쿼리에서 ifnull 사용했습니다. 불린값으로 수정해볼께요

Reply:
> 반환값이 null 안나오게 쿼리에서 ifnull 사용했습니다. 불린값으로 수정해볼께요

IFNULL은 getBeginAndEndPostNo 말고 다른 쿼리에 있는 것 같네요.

Reply:
글이 전혀 없는 경우가 없도록 파티가 생성되자마자 글이 하나 생성되게끔 하였습니다